eclipse-theia/theia
# custom-editor: fix closing dirty custom text editor issue

#### What it does

Fix the issue: Closing a dirty/modified custom editor based on CustomTextEditorProvider doesn't ask the user to save, it just saves it on close (https://github.com/eclipse-theia/theia/issues/11592)

#### How to test

1. use the cat scratch custom editor extension example from vscode (you can download it from this issue https://github.com/eclipse-theia/theia/issues/10787)
2. open the example.cscratch file
3. do some changes in the custom editor UI (add some cats by clicking on the scratch button)
4. close the dirty tab, you need to see the popup asking to save the file.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
